### PR TITLE
Add nix starter

### PR DESCRIPTION
**Pull request by matthewbauer**
This repository contains configuration for what users see when they click on the `Actions` tab.

It is not:
* A playground to try out scripts
* A place for you to create a workflow for your repository

---

Thank you for sending in this pull request. Please make sure you take a look at the [contributing file](https://github.com/actions/starter-workflows/blob/main/CONTRIBUTING.md). Here's a few things for you to consider in this pull request.

Please **add to this description** at the bottom :point_down:

- [x] A good description of the workflow at the bottom of this page.
- [x] Some links to the language or tool will be nice (unless its really obvious)

In the workflow and properties files:

- [X] The workflow filename of CI workflows should be the name of the language or platform, in lower case.  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").

  The workflow filename of publishing workflows should be the name of the language or platform, in lower case, followed by "-publish".
- [X] Includes a matching `ci/properties/*.properties.json` file.
- [X] Use sentence case for the names of workflows and steps, for example "Run tests".
- [X] The name of CI workflows should only be the name of the language or platform: for example "Go" (not "Go CI" or "Go Build")
- [X] Include comments in the workflow for any parts that are not obvious or could use clarification.
- [X] CI workflows should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [X] Packaging workflows should run on `release` with `types: [ created ]`.

Some general notes:

- [X] This workflow must only use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [X] This workflow must only use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  Workflows using these actions must reference the action using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [X] This workflow must not send data to any 3rd party service except for the purposes of installing dependencies.
- [X] This workflow must not use a paid service or product.

This is a simpler workflow for building a project that has Nix setup. It runs the `packages` and `checks` of a flake.nix.

Homepage: https://nixos.org/nix/

### CODE REVIEW

File: ci/nix.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: cachix/install-nix-action@v10
+
+    - name: Build Flakes
+      run: nix-build -E '(import (fetchTarball https://github.com/edolstra/flake-compat/archive/master.tar.gz) { src = ./.; }).defaultNix.packages.${builtins.currentSystem}'
```
Comment (jtojnar) [inline ci/nix.yml:17]: Flakes are still experimental technology that can change any time so I would be wary of promoting it to people who did not opt in into that.

**Comment by hiloboi** [comment]
A good description of the workflow at the bottom of this page.
 Some links to the language or tool will be nice (unless its really obvious)

File: ci/properties/nix.properties.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "name": "Nix",
+    "description": "Run tests for a Nix Flake",
```
Comment (jtojnar) [inline ci/properties/nix.properties.json:3]: No longer doing anything with flakes.

**Comment by matthewbauer** [comment]
Updated with docs

**Comment by andymckay** [comment]
We've got a starter workflow over here: https://github.com/actions/starter-workflows/pull/580 and I suspect we could do with one instead of two. One further question: what license is the nix logo under, this repository is under the MIT license https://github.com/actions/starter-workflows/blob/main/LICENSE

**Comment by andymckay** [comment]
One more, from the pull request template 👇🏾 

This workflow must only use actions that are produced by the language or ecosystem that the workflow supports. These actions must be published to the GitHub Marketplace. Workflows using these actions must reference the action using the full 40 character hash of the action's commit instead of a tag. Additionally, workflows must include the following comment at the top of the workflow file:

```
# This workflow uses actions that are not certified by GitHub.
# They are provided by a third-party and are governed by
# separate terms of service, privacy policy, and support
# documentation.
```

**Comment by matthewbauer** [comment]
Ok, closing in favor of https://github.com/actions/starter-workflows/pull/580

**Review by Anonymousx-legon — COMMENTED** [review]
Actually this one was done by someone else who accessed my account. But I am making sure that I have everything secure now. And I will be working hard for this community.  I deeply apologize for some repositories I haven't actually made. But I do hold responsibility. 

Thank you for fixing it. Great job!

**Review by Newman1515 — APPROVED** [review]
Yoo

**Pull request closed without merging**